Intermittent debug(T) test failure: java.lang.AssertionError: Verification failed: call 5 of 5: class org.mozilla.fenix.components.AnalyticsKt.isSentryEnabled()) should not be called

Failed 3/5 runs.

@pocmo Please check the recent commits if any of those triggers this.

Log: https://treeherder.mozilla.org/logviewer?job_id=324674867&repo=fenix

```
SUITE: org.mozilla.fenix.shortcut.PwaOnboardingObserverTest
  TEST: GIVEN cfr should not yet be shown WHEN installable page is loaded THEN counter is incremented
  SUCCESS

  TEST: GIVEN cfr should be shown WHEN installable page is loaded THEN we navigate to onboarding fragment
  SUCCESS

  TEST: GIVEN web app is not installable WHEN page with manifest is loaded THEN nothing happens
  FAILURE

java.lang.AssertionError: Verification failed: call 5 of 5: class org.mozilla.fenix.components.AnalyticsKt.isSentryEnabled()) should not be called

Calls:
1) class org.mozilla.fenix.components.AnalyticsKt.isSentryEnabled()


Stack traces:
1)                                                                 io.mockk.impl.InternalPlatform.captureStackTrace              (InternalPlatform.kt:121)           
                                                                     io.mockk.impl.stub.MockKStub.handleInvocation               (MockKStub.kt:247)                  
                                   io.mockk.impl.instantiation.JvmMockFactoryHelper$mockHandler$1.invocation                     (JvmMockFactoryHelper.kt:25)        
                                                            io.mockk.proxy.jvm.advice.Interceptor.call                           (Interceptor.kt:20)                 
                                                         org.mozilla.fenix.components.AnalyticsKt.isSentryEnabled                (Analytics.kt:134)                  
                                                            org.mozilla.fenix.ext.NavControllerKt.recordIdException              (NavController.kt:31)               
                                                            org.mozilla.fenix.ext.NavControllerKt.nav                            (NavController.kt:22)               
                                                            org.mozilla.fenix.ext.NavControllerKt.nav$default                    (NavController.kt:18)               
        org.mozilla.fenix.shortcut.PwaOnboardingObserver$start$1$invokeSuspend$$inlined$collect$1.emit                           (Collect.kt:138)                    
      mozilla.components.support.ktx.kotlinx.coroutines.flow.FlowKt$ifChanged$$inlined$filter$1$2.emit                           (Collect.kt:136)                    
   org.mozilla.fenix.shortcut.PwaOnboardingObserver$start$1$invokeSuspend$$inlined$mapNotNull$1$2.emit                           (Collect.kt:136)                    
                                                       kotlinx.coroutines.flow.FlowKt__ChannelsKt.emitAllImpl$FlowKt__ChannelsKt (Channels.kt:58)                    
                                         kotlinx.coroutines.flow.FlowKt__ChannelsKt$emitAllImpl$1.invokeSuspend                  (Channels.kt:-1)                    
                                              kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith                     (ContinuationImpl.kt:33)            
                                                                kotlinx.coroutines.DispatchedTask.run                            (DispatchedTask.kt:56)              
                                                  kotlinx.coroutines.test.TestCoroutineDispatcher.dispatch                       (TestCoroutineDispatcher.kt:50)     
                                              kotlinx.coroutines.test.internal.TestMainDispatcher.dispatch                       (MainTestDispatcher.kt:35)          
                                                              kotlinx.coroutines.DispatchedTaskKt.dispatch                       (DispatchedTask.kt:106)             
                                                   kotlinx.coroutines.CancellableContinuationImpl.dispatchResume                 (CancellableContinuationImpl.kt:308)
                                                   kotlinx.coroutines.CancellableContinuationImpl.completeResume                 (CancellableContinuationImpl.kt:395)
                                       kotlinx.coroutines.channels.AbstractChannel$ReceiveElement.completeResumeReceive          (AbstractChannel.kt:872)            
                                                     kotlinx.coroutines.channels.ConflatedChannel.offerInternal                  (ConflatedChannel.kt:61)            
                                                  kotlinx.coroutines.channels.AbstractSendChannel.send                           (AbstractChannel.kt:133)            
                                                     kotlinx.coroutines.channels.ChannelCoroutine.send$suspendImpl               (ChannelCoroutine.kt:-1)            
                                                     kotlinx.coroutines.channels.ChannelCoroutine.send                           (ChannelCoroutine.kt:-1)            
                       mozilla.components.lib.state.ext.StoreExtensionsKt$flow$1$subscription$1$1.invokeSuspend                  (StoreExtensions.kt:169)            
                                              kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith                     (ContinuationImpl.kt:33)            
                                                                kotlinx.coroutines.DispatchedTask.run                            (DispatchedTask.kt:56)              
                                                             kotlinx.coroutines.EventLoopImplBase.processNextEvent               (EventLoop.common.kt:274)           
                                                             kotlinx.coroutines.BlockingCoroutine.joinBlocking                   (Builders.kt:84)                    
                                                        kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking                    (Builders.kt:59)                    
                                                                    kotlinx.coroutines.BuildersKt.runBlocking                    (-:1)                               
                                                        kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking$default            (Builders.kt:38)                    
                                                                    kotlinx.coroutines.BuildersKt.runBlocking$default            (-:1)                               
                         mozilla.components.lib.state.ext.StoreExtensionsKt$flow$1$subscription$1.invoke                         (StoreExtensions.kt:169)            
                         mozilla.components.lib.state.ext.StoreExtensionsKt$flow$1$subscription$1.invoke                         (StoreExtensions.kt:-1)             
                                                  mozilla.components.lib.state.Store$Subscription.dispatch$lib_state_release     (Store.kt:152)                      
                                                               mozilla.components.lib.state.Store.transitionTo$lib_state_release (Store.kt:105)                      
                          mozilla.components.lib.state.internal.ReducerChainBuilder$build$chain$1.invoke                         (ReducerChainBuilder.kt:52)         
                          mozilla.components.lib.state.internal.ReducerChainBuilder$build$chain$1.invoke                         (ReducerChainBuilder.kt:18)         
                    mozilla.components.lib.state.internal.ReducerChainBuilder$build$threadCheck$1.invoke                         (ReducerChainBuilder.kt:57)         
                    mozilla.components.lib.state.internal.ReducerChainBuilder$build$threadCheck$1.invoke                         (ReducerChainBuilder.kt:18)         
        mozilla.components.lib.state.internal.ReducerChainBuilder$build$$inlined$forEach$lambda$1.invoke                         (ReducerChainBuilder.kt:62)         
        mozilla.components.lib.state.internal.ReducerChainBuilder$build$$inlined$forEach$lambda$1.invoke                         (ReducerChainBuilder.kt:18)         
                                                    mozilla.components.lib.state.Store$dispatch$1.invokeSuspend                  (Store.kt:91)                       
                                              kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith                     (ContinuationImpl.kt:33)            
                                                                kotlinx.coroutines.DispatchedTask.run                            (DispatchedTask.kt:56)              
                                                          java.util.concurrent.ThreadPoolExecutor.runWorker                      (ThreadPoolExecutor.java:1149)      
                                                   java.util.concurrent.ThreadPoolExecutor$Worker.run                            (ThreadPoolExecutor.java:624)       
                                                                                 java.lang.Thread.run                            (Thread.java:748)                   
	at io.mockk.impl.recording.states.VerifyingState.failIfNotPassed(VerifyingState.kt:66)
	at io.mockk.impl.recording.states.VerifyingState.recordingDone(VerifyingState.kt:42)
	at io.mockk.impl.recording.CommonCallRecorder.done(CommonCallRecorder.kt:47)
	at io.mockk.impl.eval.RecordedBlockEvaluator.record(RecordedBlockEvaluator.kt:60)
	at io.mockk.impl.eval.VerifyBlockEvaluator.verify(VerifyBlockEvaluator.kt:30)
	at io.mockk.MockKDsl.internalVerify(API.kt:118)
	at io.mockk.MockKKt.verify(MockK.kt:146)
	at io.mockk.MockKKt.verify$default(MockK.kt:143)
	at org.mozilla.fenix.shortcut.PwaOnboardingObserverTest.GIVEN web app is not installable WHEN page with manifest is loaded THEN nothing happens(PwaOnboardingObserverTest.kt:103)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:55)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecutor.runTestClass(JUnitTestClassExecutor.java:110)
	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecutor.execute(JUnitTestClassExecutor.java:58)
	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecutor.execute(JUnitTestClassExecutor.java:38)
	at org.gradle.api.internal.tasks.testing.junit.AbstractJUnitTestClassProcessor.processTestClass(AbstractJUnitTestClassProcessor.java:62)
	at org.gradle.api.internal.tasks.testing.SuiteTestClassProcessor.processTestClass(SuiteTestClassProcessor.java:51)
	at sun.reflect.GeneratedMethodAccessor121.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:36)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.internal.dispatch.ContextClassLoaderDispatch.dispatch(ContextClassLoaderDispatch.java:33)
	at org.gradle.internal.dispatch.ProxyDispatchAdapter$DispatchingInvocationHandler.invoke(ProxyDispatchAdapter.java:94)
	at com.sun.proxy.$Proxy2.processTestClass(Unknown Source)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.processTestClass(TestWorker.java:119)
	at sun.reflect.GeneratedMethodAccessor120.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:36)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.internal.remote.internal.hub.MessageHubBackedObjectConnection$DispatchWrapper.dispatch(MessageHubBackedObjectConnection.java:182)
	at org.gradle.internal.remote.internal.hub.MessageHubBackedObjectConnection$DispatchWrapper.dispatch(MessageHubBackedObjectConnection.java:164)
	at org.gradle.internal.remote.internal.hub.MessageHub$Handler.run(MessageHub.java:414)
	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
	at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:48)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:56)
	at java.lang.Thread.run(Thread.java:748)
```

--- a/app/src/test/java/org/mozilla/fenix/shortcut/PwaOnboardingObserverTest.kt
+++ b/app/src/test/java/org/mozilla/fenix/shortcut/PwaOnboardingObserverTest.kt
@@ -68,8 +68,10 @@ class PwaOnboardingObserverTest {
 
     @Test
     fun `GIVEN cfr should not yet be shown WHEN installable page is loaded THEN counter is incremented`() {
-        pwaOnboardingObserver.start()
         every { webAppUseCases.isInstallable() } returns true
+        every { settings.userKnowsAboutPwas } returns false
+        every { settings.shouldShowPwaCfr } returns false
+        pwaOnboardingObserver.start()
 
         store.dispatch(ContentAction.UpdateWebAppManifestAction("1", mockk())).joinBlocking()
         verify { settings.incrementVisitedInstallableCount() }
@@ -81,9 +83,10 @@ class PwaOnboardingObserverTest {
 
     @Test
     fun `GIVEN cfr should be shown WHEN installable page is loaded THEN we navigate to onboarding fragment`() {
-        pwaOnboardingObserver.start()
         every { webAppUseCases.isInstallable() } returns true
+        every { settings.userKnowsAboutPwas } returns false
         every { settings.shouldShowPwaCfr } returns true
+        pwaOnboardingObserver.start()
 
         store.dispatch(ContentAction.UpdateWebAppManifestAction("1", mockk())).joinBlocking()
         verify { settings.incrementVisitedInstallableCount() }
@@ -95,8 +98,10 @@ class PwaOnboardingObserverTest {
 
     @Test
     fun `GIVEN web app is not installable WHEN page with manifest is loaded THEN nothing happens`() {
-        pwaOnboardingObserver.start()
         every { webAppUseCases.isInstallable() } returns false
+        every { settings.userKnowsAboutPwas } returns false
+        every { settings.shouldShowPwaCfr } returns true
+        pwaOnboardingObserver.start()
 
         store.dispatch(ContentAction.UpdateWebAppManifestAction("1", mockk())).joinBlocking()
         verify(exactly = 0) { settings.incrementVisitedInstallableCount() }

